### DISCUSSION TITLE
How do you verify users who didn't click their original email?

### DISCUSSION THREAD

**Discussion by hyperknot** [ory/kratos#3036] [Q&A] [answered]
How do you verify users who didn't click their original email?

I found no /admin endpoints for re-sending verification emails. Isn't it a very common scenario to re-send verification emails?

**Comment by hyperknot** [comment]
There are some things I don't get about verification:
- Why is the normal flow UI asking for an email address? Why would anyone would like to verify someone else's email address?
- Also what confuses me even more is [this comment](https://github.com/ory/kratos/issues/2473#issuecomment-1152064929), saying that verified address and recovery address can be separate. At this point I feel lost, how can they be different?

**Comment by hyperknot** [comment]
In Firebase for example, there is 

- `auth.sendEmailVerification()` - to send the automatic email template
- `auth.generateEmailVerificationLink()` - to generate a link to be used in a custom email

**Comment by vinckr** [comment, marked as answer]
Hello @hyperknot 
Apologies for the late reply 🙇 

Browser clients API:

- https://www.ory.sh/docs/reference/api#tag/frontend/operation/createBrowserVerificationFlow

API clients API:

- https://www.ory.sh/docs/reference/api#tag/frontend/operation/createNativeVerificationFlow

```
curl -s -i -X GET \
    -H "Accept: text/html" \
    https://playground.projects.oryapis.com/ui/verification

```
For AJAX 

```
curl -s -i -X GET \
    -H "Accept: application/json" \
    https://playground.projects.oryapis.com/ui/verification | jq
```

For API clients

```
curl -s -X GET \
    -H "Accept: application/json"  \
    https://playground.projects.oryapis.com/self-service/verification/api

```

an older diagram but it still checks out(?!) (fyi @jonas-jonas)

[![](https://mermaid.ink/img/pako:eNplUMtOxDAM_BXLR7TlwLEHTvwB3AiqrMZtIpK4ShzQarX_jguHXZXbeGY8flxwFs84YlNSfom0VsrD1xO8P3zAMDzDHEQaT5k1iD9Uu9646MSZYrqHB6WIMtS4BgVZ7oW3wNAbV8h0hiWmBNIV1Ng_fZGaIZbNSFopFqAGBN9mVoHKg0X5m_0R9vJ32GGXjVpjP82BUuKy8n9id9nNeMLM1fq8_eTiCoBDy8_scDToqX46dOVqPuoqr-cy46i18wn75m8vxHGh1Pj6A_iQeqw?type=png)](https://mermaid.live/edit#pako:eNplUMtOxDAM_BXLR7TlwLEHTvwB3AiqrMZtIpK4ShzQarX_jguHXZXbeGY8flxwFs84YlNSfom0VsrD1xO8P3zAMDzDHEQaT5k1iD9Uu9646MSZYrqHB6WIMtS4BgVZ7oW3wNAbV8h0hiWmBNIV1Ng_fZGaIZbNSFopFqAGBN9mVoHKg0X5m_0R9vJ32GGXjVpjP82BUuKy8n9id9nNeMLM1fq8_eTiCoBDy8_scDToqX46dOVqPuoqr-cy46i18wn75m8vxHGh1Pj6A_iQeqw)

As for this comment, the core use case of verification is this: 

> Using this feature implements the so-called "account activation". Depending on your scenario or use cases, you may either choose to limit what an identity without verified addresses is able to do in your application logic or API Gateways, or make Ory Kratos prevent signing into accounts without verified addresses.

Of course it usually makes sense to have the recovery and verification email as the same on (since you want an active/verified email to be the recovery address)

**Reply by vinckr** [reply]
Hello @hyperknot 
Did that answer your question? I marked it as answered for now, but feel free to come back to this discussion any time.

**Reply by hyperknot** [reply]
Hi @vinckr yes, thanks it did answer. What confuses me is the whole concept of Kratos where you have to enter an email to verify it. I ended up manually modifying the flow to fill in the email.
